File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        first = strs[0]
        for word in strs[1:]:
            lcp = ""
            for i in range(min(len(first), len(word))):
                if first[i] == word[i]:
                    lcp += first[i]
                else:
                    break
            
            first = lcp
            if not first:
                break
            
        return first

File: 14-longest-common-prefix/longest-common-prefix.c
char* longestCommonPrefix(char** strs, int strsSize) {
    static char first[200];  // use static so it persists after function returns
    strcpy(first, strs[0]);

    for (int i = 1; i < strsSize; i++) {
        int j = 0;
        while (first[j] && strs[i][j] && first[j] == strs[i][j]) {
            j++;
        }
        first[j] = '\0';
        if (first[0] == '\0') {
            break;
        }
    }

    return first;
    
}